src/main/java/io/deephaven/hash/KeyedDoubleObjectHash.java

```diff
@@ -251,19 +251,25 @@ public synchronized V replace(double key, V value) {
   }
 
   public synchronized boolean replace(Double key, V oldValue, V newValue) {
+    if (oldValue == null) {
+      throw new NullPointerException("oldValue is null, but this map cannot hold null values");
+    }
     if (!doubleKeyDef.equalDoubleKey(key, newValue)) {
       throw new IllegalArgumentException(
           "key and value are inconsistent:" + key + " and " + doubleKeyDef.getDoubleKey(newValue));
     }
-    return internalPut(newValue, KeyedDoubleObjectHash.REPLACE, oldValue) != null;
+    return internalPut(newValue, KeyedDoubleObjectHash.REPLACE, oldValue).equals(oldValue);
```

Comment:
Do we want to go a step further and have `internalPut` return `null` when it found a non-matching key in mode `REPLACE`, to skip the extra equals? Or is that too much risk/complication? I think as long as we have good unit tests, I'm content to take the risk.

Reply:
I'm happy as with as little change as possible.